Breeze.TumbleBit.Client/TumbleBitManager.cs

```diff
@@ -353,29 +351,46 @@ public enum TumbleState
                     this.logger.LogDebug("Not enough masternode registrations downloaded yet: " + registrations.Count);
                     return Result.Fail<ClassicTumblerParameters>("Not enough masternode registrations downloaded yet");
                 }
-                
+
                 registrations.Shuffle();
 
-                // Since the list is shuffled, we can simply iterate through it and try each server until one is valid & reachable
+                // Since the list is shuffled, we can simply iterate through it and try each server until one is valid & reachable.
                 foreach (RegistrationRecord record in registrations)
                 {
                     this.TumblerAddress = "ctb://" + record.Record.OnionAddress + ".onion?h=" + record.Record.ConfigurationHash;
 
-                    var attemptConnection = await TryUseServer();
+                    try
+                    {
+                        var attemptConnection = await TryUseServer();
 
-                    if (!attemptConnection.Failure)
+                        if (!attemptConnection.Failure)
+                        {
+                            return attemptConnection;
+                        }
+                    }
+                    catch (Exception e)
                     {
-                        return attemptConnection;
+                        this.logger.LogDebug("Unable to connect to masternode:" + this.TumblerAddress);
                     }
                 }
 
-                // If we reach this point, no servers were reachable
+                // If we reach this point, no servers were reachable.
                 this.logger.LogDebug("Did not find a valid registration");
                 return Result.Fail<ClassicTumblerParameters>("Did not find a valid registration");
             }
             else
             {
-                return await TryUseServer();
+                var attemptConnection = await TryUseServer();
+
+                if (!attemptConnection.Failure)

```

Comment:
Same as above. I would probably also call this variable `connectionResult`. `attemptConnection` sounds like a verb, but you are getting a result from `TryUseServer`, not an action